daemons/dss-sync/app.py

```diff
@@ -24,7 +29,93 @@ def process_new_syncable_object(event, context):
     if event.get("Event") == "s3:TestEvent":
         app.log.info("DSS sync daemon received S3 test event")
     else:
-        s3 = boto3.resource("s3")
-        bucket = s3.Bucket(event['Records'][0]["s3"]["bucket"]["name"])
+        bucket = resources.s3.Bucket(event['Records'][0]["s3"]["bucket"]["name"])
         obj = bucket.Object(unquote(event['Records'][0]["s3"]["object"]["key"]))
-        sync_blob(source_platform="s3", source_key=obj.key, dest_platform="gs", logger=app.log)
+        sync_blob(source_platform="s3", source_key=obj.key, dest_platform="gs", logger=app.log, context=context)
+
+@app.sns_topic_subscriber("dss-gs-composite-upload-ready")
+def compose_upload(event, context):
+    msg = json.loads(event["Records"][0]["Sns"]["Message"])
+    gs = dss.Config.get_cloud_specific_handles("gcp")[0].gcp_client
+    gs_bucket = gs.get_bucket(msg["dest_bucket"])
+    while True:
+        try:
+            context.log("Composing, stage 1")
+            compose_stage2_blob_names = []
+            if msg["total_parts"] > 32:
```

Comment:
This is the maximum number of parts you can join at a time on GS. (There is also a 1024 part total limit, so it decomposes into two passes, 32*32).

Added a variable to hold this number, and also a docstring at the top of this function with a link to the docs.